### ISSUE TITLE
Lack of possibility to choose location on the map using a screenreader

### ISSUE THREAD

**Issue by Piciok** [BeWelcome/rox#114]
It is currently not possible for blind users who rely on screenreaders to make use of any part of the site that requires one to choose the location on a map, E.G. defining your location during registration.

Steps to reproduce:
1.navigate to any view where choosing a location using the provided map view is required;
2. Try to make the choice using a screenreader such as NVDA or Jaws.
Expected Behaviour: there should be some kind of accessible alternative;
What Happens: as map widgets like this are mostly inaccessible, it is impossible to choose anything.
Desktop
System: Windows 10
Browser: Firefox
but this will be an issue on any platform/browser combination.

Context: blind people access websites and apps on their electronic devices of choice using a screenreader, a special piece of software that presents the text shown on the screen, the controls and the context around them in the form of synthesized speech or braille output on a braille display. This solution works as long as all of the components accessed are coded in an accessible way that is all of the items are able to be recognized by this piece of software and are possible to be navigated through using a keyboard. NVDA, is a good example of a screenreader as it is free and open. Its repo is located at:
https://github.com/nvaccess/nvda
Map views that are essential in order to use all of Trustroot's features are sadly one of those elements that aren't accessible at this time. Because there is no alternative available, a blind person is unable to, among others, complete the  registration. My suggestion would be to either add an edit box that accepts geographical coordinates or a Google Maps/Mapquest link from which those can be obtained. Current location could be also used if the device supports GPS.
This minor change would make a huge difference to yet another community of travelers. I hope we can work it out.

**Comment by thisismeonmounteverest** [comment]
@piciok: There is no need to set an exact location on a map with BeWelcome. Entering a location into the dialog box should be enough.
I check what can be done to make this more obvious for people who use screenreaders.

**Comment by Piciok** [comment]
I have just checked the website again.

In the second step of the signup process, I am asked to enter my hosting location. I pick my place using the search box and the suggestions provided through which I navigate with the arrow keys. I then press enter on the result I want to have inserted, the suggestion is in the box, I press Next and I am taken back to the hosting location screen.

Is there anything I am missing?

**Comment by thisismeonmounteverest** [comment]
@piciok: Could you let me know which location you're trying? Is it possible that you disabled Javascript?

**Comment by Piciok** [comment]
Not that I am aware of it. I think it's on actually.

The location I've been trying with is Pilchowice, Silesia, Poland

**Comment by thisismeonmounteverest** [comment]
I tried that location and I'm forwarded to step 4 when I click on 'Next Step'. Could you try this on https://beta.bewelcome.org ? Because if this throws an error somewhere it's easier to find it in the logs there.

**Comment by Piciok** [comment]
Thank you for this tip. I managed to successfully complete my registration on the beta site. Should we call it a closed one then?
